packages/deployment/app/kes.override.js

```diff
@@ -401,35 +431,57 @@ class UpdatedKes extends Kes {
         resolve(JSON.parse(body).tag_name);
       });
     });
-  };
+  }
 
   /**
    * Determines which release version should be downloaded from
    * cumulus-message-adapter repository and then downloads that file.
    *
-   * @return {Promise}
+   * @returns {Promise} returns the path of the extracted message adapter
    */
   fetchMessageAdapter() {
-    const messageAdapterVersion = this.config.message_adapter_version;
-    const releaseDownloadBaseUrl = `https://github.com/${this.messageAdapterGitPath}/releases/download`;
-    const messageAdapterFilename = this.config.message_adapter_filename;
-
-    if (!messageAdapterVersion) {
-      return this.fetchLatestMessageAdapterRelease()
-        .then((latestReleaseVersion) => {
-          const releaseLocation = `${releaseDownloadBaseUrl}/${latestReleaseVersion}/${messageAdapterFilename}`;
-          return this.downloadZipfile(releaseLocation, messageAdapterFilename);
-        });
-    } else {
-      const releaseLocation = `${releaseDownloadBaseUrl}/${messageAdapterVersion}/${messageAdapterFilename}`;
-      return this.downloadZipfile(releaseLocation, messageAdapterFilename);
+    if (this.config.message_adapter_filename) {
+      const messageAdapterVersion = this.config.message_adapter_version;
+      const releaseDownloadBaseUrl = `https://github.com/${this.messageAdapterGitPath}/releases/download`;
+      const messageAdapterFilename = this.config.message_adapter_filename;
+
+      const releaseLocation = `${releaseDownloadBaseUrl}/` +
+                              `${messageAdapterVersion}/${messageAdapterFilename}`;
+      let adapterVersion = Promise.resolve(releaseLocation);
```

Comment:
This function would be easier to follow if half of it were extracted into an `adapterVersion()` function.